### PR TITLE
Feature/add lens handle identity

### PR DESCRIPTION
**Pull request by fengshanshan**

### CODE REVIEW

File: src/upstream/rss3/mod.rs [hunk 1]
```diff
@@ -217,14 +218,51 @@ async fn save_item(p: ResultItem) -> Result<TargetProcessedList, Error> {
...
 
-    let found = p
-        .actions
-        .into_iter()
-        .find(|a| a.tag == "collectible".to_string());
+    let mut found = None;
+
+    for a in p.actions.iter() {
+        if p.tag == "social" && a.tag_type == "mint" {
+            found = Some(a);
```
Comment (nykma) [inline src/upstream/rss3/mod.rs:225]: Will `found` be overwritten by another `iter()` ?

**Review by nykma — APPROVED** [review]

File: src/upstream/rss3/mod.rs (hunk 1 above)
Comment (fengshanshan) [reply src/upstream/rss3/mod.rs:225]: found declaim as None before actions iterator, it's a new variable in save item iterator process.

Comment (nykma) [reply src/upstream/rss3/mod.rs:225]: Should use `let found = p.actions.iter().find(|a| .....)` instead.

File: src/upstream/rss3/mod.rs [hunk 2]
```diff
@@ -217,14 +218,51 @@ async fn save_item(p: ResultItem) -> Result<TargetProcessedList, Error> {
         return Ok(vec![]);
     }
 
-    let found = p
-        .actions
-        .into_iter()
-        .find(|a| a.tag == "collectible".to_string());
+    let mut found = None;
```
Comment (nykma) [inline src/upstream/rss3/mod.rs:221]:
```rust
    let found = p.actions.iter().find(|a| {
        (p.tag == "social" && a.tag_type == "mint")
            || (p.tag == "collectible" && a.tag == "collectible")
    });

    if found.is_none() {
        return Ok(vec![]);
    }
    let real_action = found.unwrap();
```

File: src/upstream/rss3/mod.rs (hunk 1 above)
Comment (fengshanshan) [reply src/upstream/rss3/mod.rs:225]: good suggestion

**Review by nykma — APPROVED** [review]
:shipit:

**Pull request merged**